### PR TITLE
add `neon_local` subcommand for the broker & use that from regression tests

### PR DESCRIPTION
**Pull request by problame**
There's currently no way to just start/stop broker from `neon_local`.

This PR
* adds a sub-command
* uses that sub-command from the test suite instead of the pre-existing Python `subprocess` based approach.

Found this useful during investigation https://github.com/neondatabase/cloud/issues/16886.

### CODE REVIEW

**Review by yliang412 — APPROVED** [review]

File: test_runner/regress/test_wal_acceptor.py [hunk 1]
```diff
@@ -1406,11 +1405,7 @@ def __init__(
     ):
         self.repo_dir = repo_dir
         self.port_distributor = port_distributor
-        self.broker = NeonBroker(
-            logfile=Path(self.repo_dir) / "storage_broker.log",
-            port=self.port_distributor.get_port(),
-            neon_binpath=neon_binpath,
-        )
+        self.fake_broker_endpoint = f"http://127.0.0.1{port_distributor.get_port()}"
```
Comment (yliang412) [inline test_runner/regress/test_wal_acceptor.py:1408]: Missing a `:`

```suggestion
        self.fake_broker_endpoint = f"http://127.0.0.1:{port_distributor.get_port()}"
```

Comment (yliang412) [reply test_runner/regress/test_wal_acceptor.py:1408]: Is it the case that for `SafekeeperEnv` we don't need a real broker?

Comment (problame) [reply test_runner/regress/test_wal_acceptor.py:1408]: good catch on the missing `:`.

> Is it the case that for SafekeeperEnv we don't need a real broker?

My understanding is that this test doesn't need a pageserver (judging by name `test_safekeeper_without_pageserver`).

No need for pageserver = no need for broker.

Comment (problame) [reply test_runner/regress/test_wal_acceptor.py:1408]: I'll ask for post-merge review from Arseny, but, don't want to hold up this PR while he's on vacation

**Review by bayandin — COMMENTED** [review]
LTGM if tests pass 😄
Do we want to run benchmarks to see if there're ok?

**Review by bayandin — APPROVED** [review]

**Pull request merged**